lib/archethic/contracts/interpreter/version1/action_interpreter.ex

```diff
@@ -0,0 +1,404 @@
+defmodule Archethic.Contracts.Interpreter.Version1.ActionInterpreter do
+  @moduledoc false
+  @modules_whitelisted ["Contract"]
+
+  alias Archethic.TransactionChain.Transaction
+  alias Archethic.TransactionChain.TransactionData
+
+  alias Archethic.Contracts.Interpreter.ASTHelper, as: AST
+  alias Archethic.Contracts.Interpreter.Version1.CommonInterpreter
+  alias Archethic.Contracts.Interpreter.Version1.Library
+  alias Archethic.Contracts.Interpreter.Version1.Scope
+
+  @doc """
+  Parse the given node and return the trigger and the actions block.
+  """
+  @spec parse(any()) :: {:ok, atom(), any()} | {:error, any(), String.t()}
+  def parse({{:atom, "actions"}, _, [keyword, [do: block]]}) do
+    trigger_type = extract_trigger(keyword)
+
+    # We only parse the do..end block with the macro.traverse
+    # this help us keep a clean accumulator that is used only for scoping.
+    actions_ast = parse_block(AST.wrap_in_block(block))
+
+    {:ok, trigger_type, actions_ast}
+  catch
+    {:error, node} ->
+      {:error, node, "unexpected term"}
+
+    {:error, node, reason} ->
+      {:error, node, reason}
+  end
+
+  def parse(node) do
+    {:error, node, "unexpected term"}
+  end
+
+  @doc """
+  Execute actions code and returns either the next transaction or nil
+  """
+  @spec execute(any(), map()) :: Transaction.t() | nil
+  def execute(ast, constants \\ %{}) do
+    :ok = Macro.validate(ast)
+
+    # initiate a transaction that will be use by the "Contract" module
+    next_tx = %Transaction{data: %TransactionData{}}
+
+    # we use the process dictionary to store our scope
+    # because it is mutable.
+    #
+    # constants should already contains the "magic" variables
+    #   - "contract": current contract transaction
+    #   - "transaction": the incoming transaction (when trigger=transaction)
+    Process.put(
```

Comment:
Why do you leverage process directory instead of direct bindings in the `Code.eval_quoted` ?